set_1_basics/challenge_6_break_repeating_key_xor/decode_base64.cpp

```diff
@@ -33,26 +33,22 @@ std::istream & operator>>(std::istream & input,
         input >> base64Digit;
         if (!input) input.setstate(std::ios_base::failbit);
         manipulator.d_output[0] = decode_base64::convertFromBase64(base64Digit)
-                                  << (
-                                        decode_base64::s_numBitsInOctet
-                                        - decode_base64::s_numBitsInBase64Digit
-                                     );
+                                  << (decode_base64::s_numBitsInOctet
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Do not access "s_numBitsInOctet" with derived type "decode_base64". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS3252)